x-pack/plugins/ml/public/application/components/anomalies_table/links_menu.tsx

```diff
@@ -780,6 +803,43 @@ export const LinksMenuUI = (props: LinksMenuProps) => {
         </EuiContextMenuItem>
       );
     }
+
+    if (messageField !== null) {
+      items.push(
+        <EuiContextMenuItem
+          key="create_rule"
+          icon="machineLearningApp"
+          onClick={() => {
+            closePopover();
+            const additionalField = getAdditionalField(anomaly);
+            uiActions.getTrigger(CATEGORIZE_FIELD_TRIGGER).exec({
+              dataView: messageField.dataView,
+              field: messageField.field,
+              originatingApp: PLUGIN_ID,
+              additionalFilter: {
+                from: anomaly.source.timestamp,
+                to: anomaly.source.timestamp + anomaly.source.bucket_span * 1000,
+                ...(additionalField !== null
+                  ? {
+                      field: {
+                        name: additionalField.name,
+                        value: additionalField.value,
+                      },
+                    }
+                  : {}),
+              },
+            });
+          }}
+          data-test-subj="mlAnomaliesListRowActionPatternAnalysisButton"
+        >
+          <FormattedMessage
+            id="xpack.ml.anomaliesTable.linksMenu.patternAnalysisLabel"
+            defaultMessage="Pattern analysis"
```

Comment:
Nit - any reason you didn't go with `Run pattern analysis` here, which would be consistent with the text on the button in the dedicated view and in the field popover in Discover?

Reply:
Updated in https://github.com/elastic/kibana/pull/169400/commits/da8d015b669a35d4eda8e1732fc09fde2c5d3872